zebra/if_netlink.c

```diff
@@ -304,7 +304,7 @@ static void netlink_vrf_change(struct nlmsghdr *h, struct rtattr *tb,
 		/*
 		 * vrf_get is implied creation if it does not exist
 		 */
-		vrf = vrf_get((vrf_id_t)ifi->ifi_index,
+		vrf = vrf_get((vrf_id_t)((uint32_t)ifi->ifi_index + ((uint32_t)ns_id << 16)),
```

Comment:
We should not need to combine the namespace id and ifindex for the vrf if we are not mixing namespaces and vrf-lite.  If we are choosing one or the other than we just need to use that value specified for it.
Additionally if the ifindex of the vrf is > 16 bit ( a real possibility according to Vincent ) then this will not work as planned.